### PR TITLE
Add publish to pypi action

### PR DESCRIPTION
**Pull request by corranwebster**
This is the same as the traits-futures version.

### CODE REVIEW

File: .github/workflows/publish-on-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    - name: Build sdist and wheel
+      run: |
+        python -m build
+    - name: Publish to PyPI
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
+      run: |
+        python -m twine check --strict dist/*
+        python -m twine upload dist/*
```
Comment (mdickinson) [inline .github/workflows/publish-on-pypi.yml:31]: Nit: missing newline.

**Review by mdickinson — APPROVED** [review]
LGTM, apart from the missing newline at the end of the file.

File: .github/workflows/publish-on-pypi.yml (hunk 1 above)
Comment (corranwebster) [reply .github/workflows/publish-on-pypi.yml:31]:
```suggestion
        python -m twine upload dist/*

```

**Review by mdickinson — APPROVED** [review]
LGTM

**Pull request merged**